ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/BaseItemKeyedDatasource.kt

```diff
@@ -0,0 +1,86 @@
+package it.sysdata.ktandroidarchitecturecore
+
+import androidx.paging.*
+
+
+open class BaseItemKeyedDatasource<Key:Any, Data : Any>: ItemKeyedDataSource<Key, Data>(){
+
+    private var defaultLoadSize: Int? = null
+    protected lateinit var datas: List<Data>
+
+    override fun loadAfter(params: LoadParams<Key>, callback: LoadCallback<Data>) {
+
+        val requestedLoadSize = params.requestedLoadSize
+        val key = params.key
+
+        val item = getItemForKey(key)
+        var startPosition = datas.indexOf(item)
+        startPosition = if(startPosition < datas.size) startPosition else 0
+
+        val subList = getSubList(startPosition, requestedLoadSize)
+
+        callback.onResult(subList)
+/*      TODO: check this code
```

Comment:
remove commented part and todo or implement it.

Reply:
removed